cli/pcluster/configure/easyconfig_networking.py

```diff
@@ -0,0 +1,281 @@
+    cfn = boto3.client("cloudformation", region_name=configurer.aws_region_name)
+    try:
+        stack_name = "parallelclusternetworking-" + configurer.configuration.value.stack_name_prefix + TIMESTAMP
+        stack = cfn.create_stack(
+            StackName=stack_name,
+            TemplateURL="https://network-configuration-bucket.s3-eu-west-1.amazonaws.com/{0}".format(template_name),
+            Parameters=configurer.get_cfn_parameters(),
+            Capabilities=["CAPABILITY_IAM"],
+        )
+        LOGGER.debug("StackId: {0}".format(stack.get("StackId")))
+        LOGGER.info("Stack Name: {0}".format(stack_name))
+        status = cfn.describe_stacks(StackName=stack_name).get("Stacks")[0].get("StackStatus")
+        resource_status = ""
+        while status == "CREATE_IN_PROGRESS":
+            status = cfn.describe_stacks(StackName=stack_name).get("Stacks")[0].get("StackStatus")
+            events = cfn.describe_stack_events(StackName=stack_name).get("StackEvents")[0]
+            resource_status = (
+                "Status: %s - %s" % (events.get("LogicalResourceId"), events.get("ResourceStatus"))
+            ).ljust(80)
+            sys.stdout.write("\r%s" % resource_status)
+            sys.stdout.flush()
+            time.sleep(5)
+        # print the last status update in the logs
+        if resource_status != "":
+            LOGGER.debug(resource_status)
+
+        if status != "CREATE_COMPLETE":
+            LOGGER.critical("\nCluster creation failed.  Failed events:")
+            events = cfn.describe_stack_events(StackName=stack_name).get("StackEvents")
+            for event in events:
+                if event.get("ResourceStatus") == "CREATE_FAILED":
+                    LOGGER.info(
+                        "  - %s %s %s",
+                        event.get("ResourceType"),
+                        event.get("LogicalResourceId"),
+                        event.get("ResourceStatusReason"),
+                    )
+            LOGGER.error("Could not create the network configuration")
+            sys.exit(0)
+        print()
+        LOGGER.info("The stack has been created")
+        return cfn.describe_stacks(StackName=stack_name).get("Stacks")[0]["Outputs"]
+    except KeyboardInterrupt:
+        print()
+        LOGGER.info("Could not write up the configuration.")
+        LOGGER.info("Please check manually the created resources and stacks")
+    except Exception as e:  # Any exception is a problem
+        print()
+        LOGGER.error(
+            "An exception as occured. Please restart the configuration and check manually the created resource"
+        )
+        LOGGER.critical(e)
+        sys.exit(1)
+
+
+@handle_client_exception
+def _get_availability_zone(aws_region_name):
+    # FIXME placeholder for a function that should decide the best availability zone for the given aws_region
+    return ""
+
+
+def _choose_network_configuration(scheduler):
```

Comment:
I think the time has come to create a NetworkConfig class
and PrivateNetworkConfig and PublicPrivateNetworkConfig

with all the attributes you grouped in the dictionaries and the "create" function on it.

In this way we can reference all the items without the .value.*,
here you can just instantiate the right NetworkConfig type and in the enum you can list the two classes.